common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
+
+
+    prplmesh::hostapd::Configuration conf = load_hostapd_config(m_radio_info.iface_name);
+    if (!conf) {
+        LOG(ERROR) << "Autoconfiguration: no hostapd config to apply configuration!";
+        return false;
+    }
+
+    // If a Multi-AP Agent receives an AP-Autoconfiguration WSC message containing one or
+    // more M2, it shall validate each M2 (based on its 1905 AL MAC address) and configure
+    // a BSS on the corresponding radio for each of the M2. If the Multi-AP Agent is currently
+    // operating a BSS with operating parameters that do not completely match any of the M2 in
+    // the received AP-Autoconfiguration WSC message, it shall tear down that BSS.
+
+    // decalre a function for iterating over bss-conf and ap-vaps
+    bool abort          = false;
+    auto configure_func = [&abort, &conf, &bss_info_conf_list, &backhaul_wps_ssid,
+                           &backhaul_wps_passphrase](
+                              const std::string vap,
+                              std::list<son::wireless_utils::sBssInfoConf>::iterator bss_it) {
+        if (abort) {
+            return;
+        }
+
+        if (bss_it != bss_info_conf_list.end()) {
+
+            // we still have data on the input bss-conf list
+
+            // escape I
+            auto auth_type =
+                son::wireless_utils::wsc_to_bwl_authentication(bss_it->authentication_type);
+            if (auth_type == "INVALID") {
+                LOG(ERROR) << "Autoconfiguration: auth type is 'INVALID'; number: "
+                           << int(bss_it->authentication_type);
+                abort = true;
+                return;
+            }
+
+            // escape II
+            auto enc_type = son::wireless_utils::wsc_to_bwl_encryption(bss_it->encryption_type);
+            if (enc_type == "INVALID") {
+                LOG(ERROR) << "Autoconfiguration: enc_type is 'INVALID'; number: "
+                           << int(bss_it->encryption_type);
+                abort = true;
+                return;
+            }
+
+            // escape III
+            if (conf.get_vap_value(vap, "bssid") == "") {
+                LOG(ERROR) << "Failed to get BSSID for vap: " << vap;
+                abort = true;
+                return;
+            }
+
+            // escape IV
+            // BSS type (backhaul, fronthaul or both)
+            // Use Intel Mesh-Mode (upstream Multi-AP functionality not supported by Intel):
+            // Not supporting hybrid mode for in mesh mode (TODO - move to hybrid mode after
+            // https://github.com/prplfoundation/prplMesh/issues/889)
+            if (bss_it->fronthaul && bss_it->backhaul) {
+                LOG(ERROR) << "Not supporting hybrid VAP";
+                abort = true;
+                return;
+            }
+
+            // settings
+
+            // ssid
+            conf.set_create_vap_value(vap, "ssid", bss_it->ssid);
```

Comment:
I would move this together with all the others, so you get an overview of everything that is set.

Reply:
yes, correct, leftover.